packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/TDSCodeController.java

```diff
@@ -220,10 +224,14 @@ public ResponseEntity<Code> updateCode(@PathVariable("id") final UUID codeId, @R
 			final Optional<Code> updated = codeService.updateAsset(code);
 			return updated.map(ResponseEntity::ok)
 					.orElseGet(() -> ResponseEntity.notFound().build());
+		} catch (final NotFoundException e) {
+			log.error("Unable to find code resource", e);
+			throw new ResponseStatusException(
+					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, messages.get("code.not-found"));
 		} catch (final IOException e) {
 			log.error("Unable to update code resource", e);
 			throw new ResponseStatusException(
-					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, "Unable to update code resource");
+					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, messages.get("code.unable-to-update"));
```

Comment:
Should this be a `503 Service unavailable` instead, because the `hmi-server` did not error, we caught a service error.

Reply:
fixed